crates/cli/cli/src/commands/client/update.rs

```diff
@@ -0,0 +1,121 @@
+        required = true,
+        value_name = "CLIENT_ID",
+        help_heading = "REQUIRED",
+        help = "Identifier of the client to update"
+    )]
+    client_id: ClientId,
+
+    #[clap(
+        long = "height",
+        value_name = "HEIGHT",
+        help = "The target height of the client update. Leave unspecified for latest height."
+    )]
+    target_height: Option<u64>,
+    //
+    // #[clap(
+    //     long = "trusted-height",
+    //     value_name = "TRUSTED_HEIGHT",
+    //     help = "The trusted height of the client update. Leave unspecified for latest height."
+    // )]
+    // trusted_height: Option<u64>,
+    //
+    // #[clap(
+    //     long = "archive-address",
+    //     value_name = "ARCHIVE_ADDRESS",
+    //     group = "restart",
+    //     requires = "restart_height",
+    //     help_heading = "Update the client across a genesis restart",
+    //     help = "The RPC address of the archive node to use to fetch headers from before the restart. Requires --restart-height if used."
+    // )]
+    // archive_address: Option<String>,
+    //
+    // #[clap(
+    //     long = "restart-height",
+    //     value_name = "RESTART_HEIGHT",
+    //     group = "restart",
+    //     requires = "archive_address",
+    //     help_heading = "Update the client across a genesis restart",
+    //     help = "The height that the chain underwent a genesis restart at. Requires --archive-address if used."
+    // )]
+    // restart_height: Option<u64>,
+}
+
+impl Runnable for ClientUpdate {
+    async fn run(&self, builder: CosmosBuilder) -> Result<Output> {
+        let host_chain = builder.build_chain(&self.host_chain_id).await?;
+        let client_state = host_chain.query_client_state(&self.client_id).await?;
+        let reference_chain_id = client_state.chain_id;
+        let reference_chain = builder.build_chain(&reference_chain_id).await?;
+
+        let relayer = CanBuildRelay::build_relay(
+            &builder,
+            RelayAToBTarget,
+            &self.host_chain_id,
+            &reference_chain_id,
+            &self.client_id,
+            &self.client_id, // nothing to pass here
+        )
+        .await?;
+
+        let target_height = match self.target_height {
+            Some(height) => {
+                let height = Height::new(reference_chain_id.version(), height)
+                    .wrap_err("Invalid value for --target-height")?;
+
+                info!("Updating client using specified target height: {height}");
+                height
+            }
+            None => {
+                let height = reference_chain
+                    .with_blocking_chain_handle(|handle| {
+                        handle
+                            .query_latest_height()
+                            .map_err(|e| BaseError::relayer(e).into())
+                    })
+                    .await
+                    .wrap_err("Failed to fetch latest height on reference chain")?;
```

Comment:
We can use `CanQueryChainHeight` to query the latest chain height here.
```suggestion
                let height = reference_chain.query_chain_height().await?;
```